modin/test/exchange/dataframe_protocol/omnisci/utils.py

```diff
@@ -69,15 +70,54 @@ def export_frame(md_df, from_omnisci=False, **kwargs):
     pandas.DataFrame
     """
     if not from_omnisci:
-        return from_dataframe_to_pandas(md_df, **kwargs)
+        return from_dataframe_to_pandas_assert_chunking(md_df, **kwargs)
 
     with ForceOmnisciImport(md_df) as instance:
         md_df_exported = instance.export_frames()[0]
-        exported_df = from_dataframe_to_pandas(md_df_exported, **kwargs)
+        exported_df = from_dataframe_to_pandas_assert_chunking(md_df_exported, **kwargs)
 
     return exported_df
 
 
+def from_dataframe_to_pandas_assert_chunking(df, n_chunks=None, **kwargs):
+    """
+    Build a ``pandas.DataFrame`` from a `__dataframe__` object splitting it into `n_chunks`.
+
+    The function asserts that the `df` was split exactly into `n_chunks` before converting them to pandas.
+
+    Parameters
+    ----------
+    df : DataFrame
+        Object supporting the exchange protocol, i.e. `__dataframe__` method.
+    n_chunks : int, optional
+        Number of chunks to split `df`.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    if n_chunks is None:
+        return from_dataframe_to_pandas(df, n_chunks=n_chunks, **kwargs)
+
+    protocol_df = df.__dataframe__()
+    chunks = list(protocol_df.get_chunks(n_chunks))
+    assert len(chunks) == n_chunks
```

Comment:
why do we want to check that?..

Reply:
In order to verify that `.get_chunks(n_chunks)` doesn't cheat in difficult cases (like indivisable chunking case) and always returns the same amount of chunks as we requested.

I know we have already an [internal assertion about this](https://github.com/dchigarev/modin/blob/eebcbe6a426f6fbb26d9831134c4839b5ba1e29b/modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/dataframe.py#L325-L328) inside `.get_chunks()` implementation, but checking this twice just makes things safer :)

Reply:
What if some outside (non-Modin) implementation cheated about that? Why should we check that?..

Reply:
as this exporting function is located in our testing utils:
1. We suppose to use it on Modin frames mostly
2. We wouldn't mind if it does extra check as we don't want any kind of cheating in **testing**